modules/gdnative/include/gdnative/string.h

```diff
@@ -35,8 +35,13 @@
 extern "C" {
 #endif
 
+#include <stddef.h>
 #include <stdint.h>
-#include <wchar.h>
+
+#ifndef __cplusplus
+typedef uint32_t char32_t;
+typedef uint16_t char16_t;
```

Comment:
Why define `char16_t` at all? UTF-16 is best avoided due to some software treating those values as UCS-2.

Reply:
All Windows and some macOS API functions use UTF-16, and since we have functions to encode/decode UTF-16 in the string for this reason, why not to expose them to GDNative.